cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -103,8 +103,12 @@ private void deleteLocation(NamespaceId namespaceId) throws IOException {
     Location namespaceHome = namespacedLocationFactory.get(namespaceId.toId());
     try {
       if (hasCustomLocation(namespaceQueryAdmin.get(namespaceId.toId()))) {
-        LOG.debug("Custom location mapping %s was found while deleting namespace %s. Skipping location delete.",
-                  namespaceHome, namespaceId);
+        LOG.debug("Custom location mapping %s was found while deleting namespace %s. Deleting all data inside it but" +
+                    "skipping namespace home directory delete.", namespaceHome, namespaceId);
+        // delete everything inside the namespace home but not the namespace home as its user owned directory
+        for (Location childLocation : namespaceHome.list()) {
+          childLocation.delete(true);
```

Comment:
This is what we are doing for 3.5 for consistent  behavior. We will need to revisit this when w  do: https://issues.cask.co/browse/CDAP-6640

For data which does not belong to  cdap please see the above comment.